### DISCUSSION TITLE
"Create & Create Another" does not set focus again

### DISCUSSION THREAD

**Discussion by leo-unglaub** [filamentphp/filament#270] [Ideas]
Hey,
i am a big fan of the `Create & Create Another` feature. Super helpful for adding a lot of records quickly. I noticed on thing with this feature. When i open up the create dialogue the first time it get the focus set correctly and i can start typing and use tab to get to the next field. But when i hit `Create & Create Another` i have to use my mouse to set the focus.

Maybe it would be usefull to set the focus automatically to the first input field.
Thanks and greetings

**Comment by it-dubbelglas** [comment]
I'm also looking for this option. Did you manage to solve it yet?

**Comment by ranjeetmali** [comment]
Hi @leo-unglaub 

I recently encountered the same issue and discovered a handy trick to refocus the input field when using "Create & Create Another" in Laravel Filament.

Here’s how I handled it:

Step 1: Dispatch a custom event in afterCreate()
In your custom resource page (e.g., CreateUser.php), use the afterCreate lifecycle hook to emit a Livewire event:
```
namespace App\Filament\Resources\UserResource\Pages;

use App\Filament\Resources\UserResource;
use Filament\Resources\Pages\CreateRecord;

class CreateUser extends CreateRecord
{
    protected static string $resource = UserResource::class;

    public function afterCreate(): void
    {
        // Dispatch a custom Livewire event
        $this->dispatch('custom-event');
    }
}
```
Step 2: Listen for the event and set focus via JS
In your custom JS file (or using Filament’s @push('scripts')), listen for the event and refocus the input:

```
document.addEventListener("livewire:init", () => {
    Livewire.on("custom-event", () => {
        const element = document.getElementById("data.name");
        if (element) {
            element.focus();
        }
    });
});
```
Tip: Make sure the input has the correct ID (e.g., data.name) or adjust the selector accordingly.

Hope this helps anyone else facing the same issue!
